src/main/java/org/kairosdb/datastore/cassandra/CassandraDatastore.java

```diff
@@ -297,6 +306,19 @@ public void putDataPoint(String metricName,
                         tagNameCache.put(tagName);
                     }
                 }
+            } else if (m_cacheWarmingUpConfiguration.isEnabled()) {
+                long now = System.currentTimeMillis();
+                int interval = m_cacheWarmingUpConfiguration.getHeatingIntervalMinutes();
+                int rowSize = m_cacheWarmingUpConfiguration.getRowIntervalMinutes();
+                final long nextRowTime = rowTime + m_rowWidthWrite;
+                final DataPointsRowKey nextBucketRowKey = new DataPointsRowKey(metricName, nextRowTime, dataPoint.getDataStoreDataType(), tags);
+                if (m_cacheWarmingUpLogic.isWarmingUpNeeded(nextBucketRowKey.hashCode(), now, nextRowTime, interval, rowSize)) {
+                    final ByteBuffer serializeNextKey = DATA_POINTS_ROW_KEY_SERIALIZER.toByteBuffer(nextBucketRowKey);
+                    if (!rowKeyCache.isKnown(serializeNextKey)) {
+                        storeRowKeyReverseLookups(metricName, nextRowTime, serializeNextKey, rowKeyTtl, tags);
```

Comment:
propose to write an info-log here for the 1st insertion of a key row in next bucket cache for the actual 'row interval'